Common/src/geometry/CPhysicalGeometry.cpp

```diff
@@ -11378,9 +11382,94 @@ void CPhysicalGeometry::SetWallDistance(const CConfig *config, CADTElemClass *Wa
       unsigned long  elemID;
       int            rankID;
       su2double      dist;
+      su2double      localRoughness;
+      int index;
+
       WallADT->DetermineNearestElement(nodes->GetCoord(iPoint), dist, markerID, elemID, rankID);
+
       nodes->SetWall_Distance(iPoint, min(dist,nodes->GetWall_Distance(iPoint)));
+      index = GlobalMarkerStorageDispl[rankID]+ markerID;
+      localRoughness = GlobalRoughness_Height[index];
+      nodes->SetRoughnessHeight(iPoint, localRoughness);
     }
+    
   }
   // end SU2_OMP_PARALLEL
 }
+
+void CPhysicalGeometry::SetGlobalMarkerRoughness(CConfig *config) {
+
+unsigned short iMarker;
+#ifndef HAVE_MPI
+  
+  int *local_displ = new int [1];
```

Comment:
This variable does not seem to be used.
Also, there is hardly ever a need for `new int [1]`, you can just have an int, and if you need to pass it by pointer to a function you just take its address (`int local_displ;  functionWantsPointer(&local_displ);`)